src/NHibernate.Test/TypesTest/DateTypeTest.cs

```diff
@@ -55,36 +56,30 @@ public void ReadWriteNormal()
 		{
 			var expected = DateTime.Today.Date;
 
-			var basic = new DateClass { DateValue = expected.AddHours(1) };
-			object savedId;
-			using (ISession s = OpenSession())
-			{
-				savedId = s.Save(basic);
-				s.Flush();
-			}
-			using (ISession s = OpenSession())
-			{
-				basic = s.Get<DateClass>(savedId);
-				Assert.That(basic.DateValue, Is.EqualTo(expected));
-				s.Delete(basic);
-				s.Flush();
-			}
+			ReadWrite(expected);
 		}
 
 		[Test]
-		public void ReadWriteBaseValue()
+		public void ReadWriteMin()
+		{
+			var expected = Sfi.ConnectionProvider.Driver.MinDate;
+
+			ReadWrite(expected);
+		}
+
+		private void ReadWrite(DateTime expected)
 		{
-			var basic = new DateClass { DateValue = new DateTime(1899,1,1) };
+			var basic = new DateClass { DateValue = expected.AddHours(1) };
```

Comment:
Adding one hour does not make sense here as we only store date.

Reply:
Got screwed first too, I could have added a comment.